utopia-remix/app/models/project.server.ts

```diff
@@ -135,3 +141,52 @@ export async function getProjectAccessLevel(params: { projectId: string }): Prom
   ensure(accessLevel != null, 'Invalid access level', Status.INTERNAL_ERROR)
   return accessLevel
 }
+
+export async function listSharedWithMeProjectsAndCollaborators(params: {
+  userId: string
+}): Promise<{
+  projects: ProjectWithoutContent[]
+  collaborators: CollaboratorsByProject
+}> {
+  // 1. grab the project IDs from the collaborators where the user id matches the one passed as argument
+  const projectIds = await prisma.projectCollaborator.findMany({
+    where: { user_id: params.userId },
+    select: { project_id: true, User: true },
+  })
+
+  // 2. grab the projects from those ids
+  const projects = await prisma.project.findMany({
```

Comment:
Surely this could be done by joining the tables?

Reply:
Oh it definitely could, but not with Prisma, because of how the queries get built with select+include :(

Reply:
ha, found a way around that! [`5408225` (#5089)](https://github.com/concrete-utopia/utopia/pull/5089/commits/5408225e5f4cb6221bc6eff618fa0d7ea0875d17)

Reply:
Sweet!